support/cas-server-support-consent-webflow/src/main/java/org/apereo/cas/web/consent/CasConsentOverviewSecurityInterceptor.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package org.apereo.cas.web.consent;
+
+import java.util.List;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.apache.commons.lang3.StringUtils;
+import org.apereo.cas.CasProtocolConstants;
+import org.pac4j.core.config.Config;
+import org.pac4j.core.context.WebContext;
+import org.pac4j.core.engine.DefaultSecurityLogic;
+import org.pac4j.core.exception.HttpAction;
+import org.pac4j.springframework.web.SecurityInterceptor;
+import org.springframework.web.servlet.ModelAndView;
+import org.springframework.web.servlet.view.RedirectView;
+
+/**
+ * This is {@link CasConsentOverviewSecurityInterceptor}.
+ *
+ * @author Arnold Bergner
+ * @since 5.2.0
+ */
+public class CasConsentOverviewSecurityInterceptor extends SecurityInterceptor {
+
+    public CasConsentOverviewSecurityInterceptor(final Config config, final String clients,
+            final String authorizers) {
+
+        super(config, clients, authorizers);
+
+        final DefaultSecurityLogic secLogic = new DefaultSecurityLogic() {

```

Comment:
I remember similar code constructs such as this. Probably could use a common place with a bit of cleanup?